.gitignore

```diff
@@ -1,4 +1,574 @@
 node_modules
-dist
 .DS_Store
 js/preload/_*.js
+
```

Comment:
ok, makes sense. I've stripped down a lot of the rules related to plugins/add-ons. Please see if the latest commit can be merged.

The idea behind adding a lot of these rules are to ensure that people using different tools don't accidentally commit things and then we end up rolling that back.